src/utils/hooks/useYoutubeAPI.js

```diff
@@ -1,45 +1,34 @@
 import { useCallback, useEffect, useState } from 'react';
 
-function useYoutubeAPI(search) {
-  const [scriptLoaded, setScriptLoaded] = useState(false);
+function useYoutubeAPI(search, videoId) {
   const [clientInit, setClientInit] = useState(false);
   const [responseData, setResponseData] = useState();
 
-  // Dynamically put the script to load GAPI and listen to when it's loaded
-  useEffect(() => {
-    if (document.getElementById('gapi-script')) {
-      return setScriptLoaded(true);
-    }
-    const scriptTag = document.createElement('script');
-    scriptTag.addEventListener('load', () => setScriptLoaded(true));
-    scriptTag.src = 'https://apis.google.com/js/client.js';
-    scriptTag.id = 'gapi-script';
-    document.body.appendChild(scriptTag);
-  }, []);
-
-  // Initialize the client once GAPI is loaded.
+  // Initialize the client
   // Timeout prevents "cannot find setApiKey of undefined" error
   useEffect(() => {
-    if (scriptLoaded) {
-      setTimeout(() => {
-        try {
-          window.gapi.client.setApiKey(process.env.REACT_APP_GAPI_KEY);
-          window.gapi.client.load(
-            'https://www.googleapis.com/discovery/v1/apis/youtube/v3/rest'
-          );
-          console.log('GAPI client loaded for API');
-          // Timeout prevents "cannot find search of undefined"
-          setTimeout(() => {
-            setClientInit(true);
-          }, 450);
-        } catch (err) {
-          console.error('Error loading GAPI client for API', err);
-        }
-      }, 100);
-    }
-  }, [scriptLoaded]);
+    setTimeout(() => {
+      try {
+        window.gapi.client.setApiKey(process.env.REACT_APP_GAPI_KEY);
+        window.gapi.client.load(
+          'https://www.googleapis.com/discovery/v1/apis/youtube/v3/rest'
+        );
+        console.log('GAPI client initialized');
+        // Timeout prevents "cannot find search of undefined"
+        setTimeout(() => {
+          setClientInit(true);
+        }, 450);
+      } catch (err) {
+        console.error('Error loading GAPI client for API', err);
+      }
+    }, 100);
+  }, []);
 
   const getList = useCallback(async () => {
+    // Must not run when getting a single video details
+    if (videoId) {
+      return;
+    }
```

Comment:
I feel like these validations could be moved to the `useEffect`. If you don't want to run them when the `videoId` you could have them in:

```javascript
useEffect(() => {
  if (!videoId) {
    getList();
  } else {
    getVideo();
  }
 }, [videoId, getList, getVideo]);
 ```
 Also this centralizes the hook behavior to a single `useEffect`.